cristin-import/src/main/java/no/unit/nva/cristin/mapper/PublicationInstanceBuilder.java

```diff
@@ -0,0 +1,200 @@
+import no.unit.nva.model.instancetypes.degree.DegreeMaster;
+import no.unit.nva.model.instancetypes.degree.DegreePhd;
+import no.unit.nva.model.instancetypes.journal.FeatureArticle;
+import no.unit.nva.model.instancetypes.journal.JournalArticle;
+import no.unit.nva.model.instancetypes.journal.JournalCorrigendum;
+import no.unit.nva.model.instancetypes.journal.JournalLeader;
+import no.unit.nva.model.instancetypes.journal.JournalLetter;
+import no.unit.nva.model.instancetypes.journal.JournalReview;
+import no.unit.nva.model.instancetypes.report.ReportResearch;
+import no.unit.nva.model.pages.MonographPages;
+import no.unit.nva.model.pages.Pages;
+import no.unit.nva.model.pages.Range;
+import software.amazon.awssdk.utils.Validate;
+
+import static no.unit.nva.cristin.lambda.constants.HardcodedValues.HARDCODED_BOOK_PEER_REVIEWED;
+import static no.unit.nva.cristin.lambda.constants.HardcodedValues.HARDCODED_BOOK_TEXTBOOK_CONTENT;
+import static no.unit.nva.cristin.lambda.constants.HardcodedValues.HARDCODED_JOURNAL_PEER_REVIEWED;
+import static no.unit.nva.cristin.mapper.CristinMainCategory.isBook;
+import static no.unit.nva.cristin.mapper.CristinMainCategory.isChapter;
+import static no.unit.nva.cristin.mapper.CristinMainCategory.isJournal;
+import static no.unit.nva.cristin.mapper.CristinMainCategory.isReport;
+import static no.unit.nva.cristin.mapper.CristinSecondaryCategory.isChapterArticle;
+import static no.unit.nva.cristin.mapper.CristinSecondaryCategory.isDegreeMaster;
+import static no.unit.nva.cristin.mapper.CristinSecondaryCategory.isDegreePhd;
+import static no.unit.nva.cristin.mapper.CristinSecondaryCategory.isFeatureArticle;
+import static no.unit.nva.cristin.mapper.CristinSecondaryCategory.isJournalArticle;
+import static no.unit.nva.cristin.mapper.CristinSecondaryCategory.isJournalCorrigendum;
+import static no.unit.nva.cristin.mapper.CristinSecondaryCategory.isJournalLeader;
+import static no.unit.nva.cristin.mapper.CristinSecondaryCategory.isJournalLetter;
+import static no.unit.nva.cristin.mapper.CristinSecondaryCategory.isJournalReview;
+import static no.unit.nva.cristin.mapper.CristinSecondaryCategory.isMonograph;
+import static no.unit.nva.cristin.mapper.CristinSecondaryCategory.isResearchReport;
+
+@SuppressWarnings("PMD.GodClass")
+public class PublicationInstanceBuilder {
+
+    public static final String ERROR_CRISTIN_OBJECT_IS_NULL = "CristinObject can not be null";
+    public static final String ERROR_PARSING_SECONDARY_CATEGORY = "Error parsing secondary category";
+    public static final String ERROR_PARSING_MAIN_CATEGORY = "Error parsing main category";
+    public static final String ERROR_PARSING_MAIN_OR_SECONDARY_CATEGORIES = "Error parsing main or secondary "
+            + "categories";
+
+    private final CristinObject cristinObject;
+
+    public PublicationInstanceBuilder(CristinObject cristinObject) {
+        Validate.notNull(cristinObject, ERROR_CRISTIN_OBJECT_IS_NULL);
```

Comment:
better use Objects.requireNonNull that is native Java. Also the class you are using is not supposed to be used by end-users of the library.